etc/systemd/system/zfs-import-cache.service.in

```diff
@@ -12,7 +12,7 @@ ConditionPathExists=@sysconfdir@/zfs/zpool.cache
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStartPre=/sbin/modprobe zfs
+ExecStartPre=-/sbin/modprobe zfs
```

Comment:
In order to address @loli10K's concerns why don't we leverage the existing `--enable-linux-builtin` configure option.  When `--enable-linux-builtin=yes` then the entire `ExecStartPre` line should be removed from the service file.  This can be done relatively easily in the `Makefile.am` in a similar fashion to the existing substitution.

Reply:
that'd work, ya

Reply:
as long as we are fine with more moving parts

Reply:
It's OK by me.  Since we're already dynamically generating the service file we might as well generate what we want based off the provided configure options.

Reply:
Well, I'm willing to do it, but I have almost no experience with makefile writing.  Wasn't able to find something like it in the other makefiles either.

Reply:
I have no memory of modprobe failing to load builtins: i could only find a bug reported for busybox's modprobe applet (https://bugs.busybox.net/show_bug.cgi?id=5270) which has since been fixed, according to my Debian8 (which is already the old stable):

```
root@linux:~# grep zfs /proc/modules 
root@linux:~# modprobe zfs
root@linux:~# echo $?
0
root@linux:~# busybox modprobe zfs
root@linux:~# echo $?
0
root@linux:~# cat /sys/module/zfs/version 
0.7.0-311_g9c5167d
root@linux:~# lsb_release -a
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 8.0 (jessie)
Release:	8.0
Codename:	jessie
root@linux:~# 
```

Please also consider the use case where we have different boot options, both builtin (usually used for recovery i guess?) and with external modules, sharing the same rootfs: unfortunately, if we have to support both options, i don't think we can conditionally remove `ExecStartPre` at build time.

With that said, i dismissed my review because i don't want to keep this from being merged, i just didn't (and still don't) understand why we would need this change and wanted to express my opinion on the matter.

Reply:
I suppose the only case to allow it to fail would be for older systems with broken busyboxes (where I probably saw it).  I'm fine dropping it if desired, just know that dracut itself merged a change just like the current one (whether they should or not).

Your point about building generic userland components makes sense to me as well.

Reply:
According to my Fedora27, busybox (busybox-1.26.2-3.fc27) is still broken, a patch was posted 2012-08-12 on busybox's issue tracker but was only merged _recently_ (https://github.com/mirror/busybox/commit/803c85a20710b8d9026775f5668237fff496dc1e). Dracut doesn't seem to be using busybox's "modprobe" applet, this may be the reason i could not find any bug reported on the Fedora issue tracker.

Debian is not affected because they ship busybox with said patch backported to older versions.